tools/verifygitlog.py

```diff
@@ -76,21 +81,23 @@ def warning(err):
     if not raw_body[-1].startswith("Signed-off-by: ") or "@" not in raw_body[-1]:
         warning("Message should be signed-off")
 
-    return errors, warnings
-
 
 def run(args):
     verbose("run", *args)
     has_errors = False
     has_warnings = False
-    for sha in git_log("%h", *args):
-        errors, warnings = verify(sha)
-        has_errors |= any(errors)
-        has_warnings |= any(warnings)
-        for err in errors:
-            print("error:", err)
-        for err in warnings:
-            print("warning:", err)
+
+    if "--check-message" in args:
+        has_errors, has_warnings = run_check_message(args[-1])
```

Comment:
if you instead make this `errors, warnings = ...` then the code below can be reused instead of duplicating?

Reply:
Not without a few more changes, because the other part runs inside a `for` loop across multiple commits.

However, you're right that this code is a bit awkward. I tried to minimise changes to the other parts, but as a result it's "smelly". I'll do a bit more refactoring.

Reply:
I agree with @stinos.  Would be good to have only one location where there is `def error(...) ...` and `def warning(...) ...` and `for err in errors: print(...)`.  The messages come either from a file or from a git sha, but after that the code should be the same to process the message and print errors/warnings.

Reply:
Oh, I see you refactored in the following commit...

Reply:
The refactor is fine.  There are better things to spend time on than making this script perfect :smile: